Update FinanceAgent v1.3

minmin-intel opened the pull request:
## Description
1. Update readme with more detailed instructions on how to use UI to set up endpoints and upload docs
2. Update supervisor agent tools yaml to include research_agent as tool
3. Update test scripts to have validations on supervisor outputs in single-turn and multi-turn tests
4. Update readme with instructions on getting Finnhub and FinancialDatasets APIs and setting those env vars

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [x] Others (enhancement, documentation, validation, etc.)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

lkk12014402 reviewed (approved):
LGTM

rbrugaro reviewed (commented):
@minmin-intel
1. The diagram shows summarization as an agent, I suggest to change the color like the other tools and mention OPEA DocSum.
2. Could we add the diagram we had to show the database and vector store design (what data goes where)?
3. Should we provide more details in the README about the specific QA strategy that is implemented, also for the research agent?

minmin-intel commented on FinanceAgent/tests/test.py:
```diff
@@ -35,13 +35,19 @@ def test_worker_agent(args):
     query = {"role": "user", "messages": user_message, "thread_id": thread_id, "stream": stream}
     ret = process_request(url, query, is_stream=stream)
     print("Response: ", ret)
+    return ret
```
resolved

minmin-intel commented:
> @minmin-intel
> 
> 1. The diagram shows summarization as an agent, I suggest to change the color like the other tools and mention OPEA DocSum.
> 2. Could we add the diagram we had to show the database and vector store design (what data goes where)?
> 3. Should we provide more details in the README about the specific QA strategy that is implemented, also for the research agent?

@rbrugaro I added descriptions and diagrams for dataprep and finqa tool. I also updated the arch picture to show summarization is using docsum. I kept the color unchanged as I wanted to show that each light blue box is a microservice.

minmin-intel commented:
> @minmin-intel
> 
> 1. The diagram shows summarization as an agent, I suggest to change the color like the other tools and mention OPEA DocSum.
> 2. Could we add the diagram we had to show the database and vector store design (what data goes where)?
> 3. Should we provide more details in the README about the specific QA strategy that is implemented, also for the research agent?

@rbrugaro I added descriptions and diagrams for dataprep and finqa tool. I updated the summarization box in the arch diagram to show docsum, but kept the color unchanged as I wanted to show each light blue box is a microservice.

minmin-intel commented:
@rbrugaro I added descriptions and diagrams for dataprep and finqa tool. I updated the summarization box in the arch diagram to show docsum, but kept the color unchanged as I wanted to show each light blue box is a microservice

rbrugaro commented:
LGTM
